components/dashboard/src/workspaces/StartWorkspaceModal.tsx

```diff
@@ -4,70 +4,21 @@
-
-    const list = (selection === 'Recent' ? p.recent : p.examples).map((e, i) =>
-        <a key={`item-${i}-${e.title}`} href={e.startUrl} className="rounded-xl group hover:bg-gray-100 dark:hover:bg-gray-800 flex p-4 my-1">
-            <div className="w-full">
-                <p className="text-base text-gray-800 dark:text-gray-200 font-semibold">{e.title}</p>
-                <p>{e.description}</p>
-            </div>
-        </a>);
-
-    return <Modal onClose={p.onClose} visible={p.visible}>
+export function StartWorkspaceModal(props: StartWorkspaceModalProps) {
+    return <Modal onClose={props.onClose} onEnter={() => false} visible={props.visible}>
         <h3 className="pb-2">New Workspace</h3>
         {/* separator */}
-        <div className="border-t border-gray-200 dark:border-gray-800 mt-2 -mx-6 px-6 pt-2">
-            <div className="flex">
-                <TabMenuItem name='Recent' selected={selection === 'Recent'} onClick={() => setSelection('Recent')} />
-                {p.examples.length>0 && <TabMenuItem name='Examples' selected={selection === 'Examples'} onClick={() => setSelection('Examples')} />}
-            </div>
-        </div>
-        <div className="border-t border-gray-200 dark:border-gray-800 -mx-6 px-6 py-2">
-            {list.length > 0 ?
-                <p className="my-4 text-base">
-                    {selection === 'Recent' ?
-                        'Create a new workspace using the default branch.' :
-                        'Create a new workspace using an example project.'}
-                </p> : <p className="h-6 my-4"></p>}
-            <div className="space-y-2 mt-4 overflow-y-scroll h-80 pr-2">
-                {list.length > 0 ? list :
-                    (selection === 'Recent' ?
-                        <div className="flex flex-col pt-10 items-center px-2">
-                            <h3 className="mb-2 text-gray-500 dark:text-gray-400">No Recent Projects</h3>
-                            <p className="text-center">Projects you use frequently will show up here.</p>
-                            <p className="text-center">Prefix a Git repository URL with {window.location.host}/# or start with an example.</p>
-                            <button onClick={() => setSelection('Examples')} className="font-medium mt-8">Select Example</button>
-                        </div> :
-                        <div className="flex flex-col pt-10 items-center px-2">
-                            <h3 className="mb-2 text-gray-500 dark:text-gray-400">No Example Projects</h3>
-                            <p className="text-center">Sorry there seem to be no example projects, that work with your current Git provider.</p>
-                        </div>)
-                }
-            </div>
+        <div className="border-t border-gray-200 dark:border-gray-800 mt-2 -mx-6 px-6 pt-4">
+            <h4 className="text-base">Search or paste a repository URL</h4>
+            <RepositoryFinder />
```

Comment:
**thought**: Hopefully it's now more clear to everyone that listing recent repositories, workspaces, etc when triggering the modal for the first time makes sense! However, sounds also ok to leave this out of the scope of this iteration.

Reply:
Sorry, I'm not sure I understand this. Would you like us to make it clearer what is being displayed / what a user should type? (I agree that presenting a user with a searchbar and nothing else can be a bit confusing at first -- I wonder how VS Code solves this with their Command Palette input 🤔)

Reply:
I don't have a concrete proposal yet but I meant to say that populating the empty modal white space, when users haven't typed anything yet, with recent repositories or workspaces could be useful.

Also:
> However, sounds also ok to leave this out of the scope of this iteration.

Reply:
Concretely, the challenge is that the new help text in the UI does not help users to understand what is being  auto-suggested or searched.

I don't think we should introduce any magic search syntax, but it would be nice to convey that the suggestions are coming from the 3 possible sources (also mentioned in the description above). This can happen either in the results themselves (indicate which result is from which source), or, my preference, in some help text below the input. It's fine to replace that when the results show up

The original suggested help text was

> Paste a repository context URL or start typing to see suggestions. [Learn more](url)

I propose adding the following text (below the input) and remove the help text above the input. 

>  Paste a repository [context URL](url) 
>  Or start typing to see suggestions from
>  ~ Your recent repositories
>  ~ Your repositories from connected integrations
>  ~ Example repositories

---

For reference here's how github does it 

**👇🏼** in in the file finder (T in code)
![Screenshot 2022-01-24 at 09 58 24](https://user-images.githubusercontent.com/849592/150784657-f69d1b93-a538-417f-b0b7-31237648ff3c.png)

**👇🏼** in the new command palette (cmd+k)

![Screenshot 2022-01-24 at 09 56 23](https://user-images.githubusercontent.com/849592/150785048-cfdbd360-588d-45d3-bc20-75abde63f4f5.png)

**👇🏼** in the new beta project command palette (cmd+k in project).

![Screenshot 2022-01-24 at 09 54 42](https://user-images.githubusercontent.com/849592/150785127-4dc4f3c1-cf5c-4b20-9eef-a98914f7e135.png)